enhancements/olm/service-account-deprecation.md

```diff
@@ -0,0 +1,262 @@
+kind: ClusterExtension
+metadata:
+  name: argocd-extension
+spec:
+  installNamespace: argocd
+  packageName: argocd-operator
+  version: 0.6.0
+  # serviceAccount is now deprecated and has no effect
+```
+
+### Controller Logic Changes
+
+- Remove all logic that: 
+  - Token Acquisition: Eliminate use of TokenRequest API to fetch short-lived tokens for user-provided ServiceAccounts.
+  - Rest Config Mapping: Remove the ServiceAccountRestConfigMapper, which dynamically generated rest.Config objects for impersonation.
+  - Synthetic Permissions: Remove conditional logic for SyntheticPermissions that depended on impersonated clients.
+- The controller now uses a static `rest.Config` created from its own identity (its default ServiceAccount). 
+- The RestConfigMapper will now be directly set to `ClusterAdminRestConfigMapper(mgr.GetConfig())`, i.e. the controller always uses its own identity (cluster-admin level config) for all reconciliation and watching operations.
+This config is passed to the helm.ActionConfigGetter, ensuring that all Helm operations (install, upgrade, uninstall) use the same identity.
+
+- Log a deprecation warning if `.spec.serviceAccount` is set
+```
+[DEPRECATION] 'spec.serviceAccount' is specified in ClusterExtension 'foo', but is ignored and will be removed in a future release.
+```
+
+## Design Details
+
+### Test Plan
+
+To validate the safe deprecation and eventual removal of the `.spec.serviceAccount` field, the following implementation and testing steps will be followed.
+
+#### Unit Tests
+  - Remove tests that exercised impersonation logic and ServiceAccount specific behavior.
+  - Add new tests to confirm:
+    - The `.spec.serviceAccount` field is ignored during reconciliation.
+    - A warning is logged when the field is set.
+    - Default behavior uses the controller’s own identity.
+
+#### E2E Tests
+- Update E2E tests to:
+  - Deploy `ClusterExtension` resources with and without the deprecated field.
+  - Verify that:
+    - Reconciliation succeeds in both cases.
+    - Deprecation warnings appear in controller logs.
+
+#### Upgrade Tests
+- Add upgrade test scenarios to validate:
+  - CRD schema updates (required -> optional -> removed) behave safely and correctly across releases.
+  - Manifests with the deprecated field do not block upgrades to 4.21 or 4.22.
+  - Manifests with the field are rejected cleanly in 4.23 (removal release).
+  - No reconciliation failures occur due to field removal when manifests are properly cleaned.
+
+### Graduation Criteria / Deprecation Plan
+
+We will deprecate the .spec.serviceAccount field over the course of three OpenShift releases following the kubernetes' deprecation policy:
+
+OpenShift 4.20 (current release): 
+- No changes. The .spec.serviceAccount field continues to be a required field and function as it currently does.
+
+OpenShift 4.21: (next release - N)
+- Mark the field as optional in the API.
+- The controller ignores the field entirely.
```

Comment:
How does this change the behaviour for an end user? If they want to leverage the older behaviour still, are they able to do so? Is there a replacement field?